samples/openapi3/server/petstore/spring-boot-oneof/src/main/java/org/openapitools/model/CreateParkAnimalCarerPersonResponsibleForOneOf.java

```diff
@@ -0,0 +1,34 @@
+package org.openapitools.model;
+
+import java.net.URI;
+import java.util.Objects;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import org.openapitools.jackson.nullable.JsonNullable;
+import java.time.OffsetDateTime;
+import javax.validation.Valid;
+import javax.validation.constraints.*;
+import io.swagger.v3.oas.annotations.media.Schema;
+
+
+import java.util.*;
+import javax.annotation.Generated;
+
+
+@JsonIgnoreProperties(
+  value = "type", // ignore manually set type, it will be automatically generated by Jackson during serialization
+  allowSetters = true // allows the type to be set during deserialization
+)
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type", visible = true)
+@JsonSubTypes({
+  @JsonSubTypes.Type(value = Cat.class, name = "CAT"),
+  @JsonSubTypes.Type(value = Dog.class, name = "NESTED_TWO"),
+  @JsonSubTypes.Type(value = Cat.class, name = "Cat"),
```

Comment:
I am also not a big fan of these JsonSubTypes being generated on top of the defined discriminator. I think this goes against the OAS specification when discriminator is present. I might invest some further time to ensure cleaner model generation (not part of this MR).